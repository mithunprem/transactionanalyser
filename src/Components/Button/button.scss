.button {
  background-color: gray;
  color: white;
  border: 2px solid gray;
  appearance: none;
  box-sizing: border-box;
  padding: 8px;
  text-transform: uppercase;
  box-shadow: none;
  border-radius: 5px;
  font-size: 14px;
  font-weight: 600;
  letter-spacing: 2.7px;
  text-align: center;
  width: 100%;
}
