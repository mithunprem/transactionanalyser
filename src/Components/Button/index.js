import React, { Fragment  } from 'react';
import PropTypes from 'prop-types';

import './button.scss';

const Button = ({ type = "text", label, onClick = () => { } }) => {
  return (
    <Fragment>
      <button className="mt-3 button" type={type} onClick={onClick}>
        {label}
      </button>
    </Fragment>
  )
}

Button.propTypes = {
  type: PropTypes.string,
  label: PropTypes.string,
  onClick: PropTypes.func
}

export default Button;
