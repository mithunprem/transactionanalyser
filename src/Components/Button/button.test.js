import React from 'react';
import { create } from 'react-test-renderer';
import Button from '../../Components/Button';

describe("Button", () => {

  const onClick = () => {};

  test("should render without throwing an error", () => {
    const button = create(<Button type="submit" label="Test" onClick={onClick}/>).toJSON()
    expect(button).toMatchSnapshot()
  });

  test("should render a disabled button when a disabled prop is passed to the component", () => {
    const button = create(<Button type="submit" label="Test" disabled={true} onClick={onClick} />).toJSON()
    expect(button).toMatchSnapshot()
  });

});
