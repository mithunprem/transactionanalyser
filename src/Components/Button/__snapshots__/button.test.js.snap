// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render a disabled button when a disabled prop is passed to the component 1`] = `
<button
  className="mt-3 button"
  disabled={true}
  onClick={[Function]}
  type="submit"
>
  Test
</button>
`;

exports[`Button should render without throwing an error 1`] = `
<button
  className="mt-3 button"
  onClick={[Function]}
  type="submit"
>
  Test
</button>
`;
